### DISCUSSION TITLE
People Search KQL to hide users that have been disabled or hidden from the address list

### DISCUSSION THREAD

**Discussion by clmmlc** [microsoft-search/pnp-modern-search#2154] [Q&A] [answered]
I am trying to create a KQL query template that will remove people results that have the attribute value msexchhidefromaddresslists="$true"

I have had look at https://www.techmikael.com/2018/01/excluding-sharepoint-online-service.html and https://docs.microsoft.com/en-us/sharepoint/dev/general-development/keyword-query-language-kql-syntax-reference#kql-operators-for-complex-queries 

So far I have tried quite a few variations on {searchTerms} (-msexchhidefromaddresslists:"true")

But this doesn't omit the hidden accounts. I am very new to this so any help would be greatly appreciated.

**Comment by wobba** [comment, marked as answer]
Did you try `NOT SPS-HideFromAddressLists:1`?

**Reply by clmmlc** [reply]
Thanks so much - this is on the right track I think but it eliminate all but 9 of our staff (we have a few hundred
![2022-06-30_13-48-28](https://user-images.githubusercontent.com/107613292/176602534-346f2da2-0406-47ba-bc2d-48ea57dfc494.png)
)

**Reply by wobba** [reply]
Then you have to confirm if these profiles has that value set, or you can use any other property solving your needs and filter on that.

**Reply by clmmlc** [reply]
We are currently managing the msExchHideFromAddressListattribute in AD and I see that it maps to SPS-HideFromAddressLists

When I add _NOT SPS-HideFromAddressLists:1_ to the query template 9 results only appear

The 9 results that appear have the msExchHideFromAddressList = not set
![2022-07-07_12-44-14](https://user-images.githubusercontent.com/107613292/177693956-7e4f3ce1-94f6-4986-b88e-895c47fc9632.png)
 
There are more than 9 profiles in our system that have this attribute = not set

There is a test group of 20 users that have this attribute set = FALSE. These are not showing in the results

There are 600 users that have the attribute set = TRUE. These are also not showing

So it seems like _NOT SPS-HideFromAddressLists:1_ isn't showing the right results

Is there something else that I am missing that needs to be set?

**Reply by wobba** [reply]
Not fully aware how the setting is sync'd over to UPA to be picked up.. that's where you should check.

**Reply by clmmlc** [reply]
Thank you :)  The user data is correctly being pushed into the SharePoint user database and that a PNP query can see it. SharePoint to UPA appears to be fine. 

Is the fact that SPS-HideFromAddressLists from the UPA is not synced to SharePoint sites the problem?
https://docs.microsoft.com/en-us/sharepoint/user-profile-sync


I also see that there are issues with the msoft people vertical not respecting respect SPS-HideFromAddressLists
(msExchHideFromAddressLists). Is this perhaps the same for the pnp modern search vertical?

https://techcommunity.microsoft.com/t5/microsoft-sharepoint-blog/exclude-users-from-delve-and-sharepoint-online-people-search/ba-p/170731

https://vladilen.com/sharepoint/hide-non-personal-account-from-microsoft-365-search/

**Reply by SinghGurd** [reply]
Make sure you set this value using AD tools and not Exchange tools. Also, try to check if mailNickName property is set too. This article [explains it nicely](https://techcommunity.microsoft.com/t5/microsoft-sharepoint-blog/exclude-users-from-delve-and-sharepoint-online-people-search/ba-p/170731) and you can test it using Delve too.

**Reply by clmmlc** [reply]
Thanks for that. We have it working nicely using AD and hidden from Delve/GAL already. It's the PNP search results web part that is not respecting the values we have set in AD I'm afraid.

**Reply by clmmlc** [reply]
If anyone has any example of a query that currently works to exclude hidden users (SPS-HideFromAddressLists:1) I would greatly appreciate it! I cannot seem to write a query that works at all and I can't roll out our search center until this is resolved.

**Reply by clmmlc** [reply]
I have realized the issue is with the query. I was trying to EXCLUDE using NOT SPS-HideFromAddressLists:1
However I should have been INCLUDE results by using "sps-hidefromaddresslists":0
Thank you so much to everyone that helped!
